client/dashboard/me/security-social-logins/apple-login.tsx

```diff
@@ -0,0 +1,104 @@
+import config from '@automattic/calypso-config';
+import { loadScript } from '@automattic/load-script';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { MouseEvent, useCallback, useEffect, useState } from 'react';
+import { getRedirectUri, getSocialServiceResponse } from './utils';
+import type { SocialLoginButtonProps, AppleClient } from './types';
+
+const appleClientUrl =
+	'https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js';
+
+// This component supports only Social Login from Apple.
+// It is used only in the Security page for now to connect and disconnect Apple accounts.
+// We can extend other flows in the future.
+export default function AppleLogin( {
+	isConnected,
+	responseHandler,
+	handleDisconnect,
+	isLoading,
+}: SocialLoginButtonProps ) {
+	const [ appleClient, setAppleClient ] = useState< AppleClient | null >( null );
+	const [ showLoading, setShowLoading ] = useState( false );
+
+	const socialServiceResponse = getSocialServiceResponse();
```

Comment:
Should we `useMemo` this one? dependencies of this variable always get triggered every time this function is called. Unless that is what we want.